src/helpers/GeneralContext.tsx

```diff
@@ -0,0 +1,52 @@
+import React, { useEffect, useState } from 'react';
```

Comment:
It's not a helper, it is a context. I believe, that should be a separate folder for it like `store` or something.